jdbc-source/src/test/java/org/springframework/cloud/stream/module/jdbc/source/JdbcSourceIntegrationTests.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.module.jdbc.source;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.hamcrest.Matchers;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.jdbc.EmbeddedDataSourceConfiguration;
+import org.springframework.boot.test.IntegrationTest;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.cloud.stream.annotation.Bindings;
+import org.springframework.cloud.stream.messaging.Source;
+import org.springframework.cloud.stream.test.binder.MessageCollector;
+import org.springframework.jdbc.core.JdbcOperations;
+import org.springframework.messaging.Message;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+/**
+ * Integration Tests for JdbcSource. Uses hsqldb as a (real) embedded DB.
+ *
+ * @author Thomas Risberg
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = {JdbcSourceApplication.class, EmbeddedDataSourceConfiguration.class})
+@IntegrationTest({"server.port=-1"})
```

Comment:
Well, to avoid extra testing time we should avoid `@EnableAutoConfiguration`, which scans the classpath and populate a lot of stuff, which is, right, useless for the particular test-case.
But... Do we have other choice?

I'll check in the debug if `@IntegrationTest({"server.port=-1"})` helps to avoid starting Embedded Tomcat. 
Right now I don't believe: we have it in the CP, so `EmbeddedServletContainerAutoConfiguration` does its magic.